flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/UnboundedEventTimeOverProcessFunction.scala

```diff
@@ -233,30 +248,28 @@ class UnboundedEventTimeRowsOverProcessFunction(
     lastAccumulator: Row,
     out: Collector[Row]): Unit = {
 
-    var j = 0
     var i = 0
-    while (j < curRowList.size) {
-      val curRow = curRowList.get(j)
-      i = 0
+    while (i < curRowList.size) {
+      val curRow = curRowList.get(i)
 
+      var j = 0
       // copy forwarded fields to output row
-      while (i < forwardedFieldCount) {
-        output.setField(i, curRow.getField(i))
-        i += 1
+      while (j < forwardedFieldCount) {
```

Comment:
I think it would be good if we could move everything (or as much as possible) related to output construction to the code-gen'd function.